File: app/src/main/java/cn/mycommons/androidmodular/app/ModuleConfig.java
package cn.mycommons.androidmodular.app;

import android.app.Activity;
import android.support.annotation.NonNull;

import java.util.HashMap;
import java.util.Map;

import cn.mycommons.modulebase.IModuleConfig;
import cn.mycommons.modulebase.IRouterProcess;

/**
 * ModuleConfig <br/>
 * Created by xiaqiulei on 2017-05-14.
 */
public class ModuleConfig implements IModuleConfig {

    @NonNull
    private final Map<String, Class<? extends Activity>> activityRouterConfig;
    @NonNull
    private final Map<String, IRouterProcess> routerProcessConfig;
    @NonNull
    private final Map<Class<?>, Class<?>> serviceConfig;

    ModuleConfig() {
        activityRouterConfig = new HashMap<>();
        routerProcessConfig = new HashMap<>();
        serviceConfig = new HashMap<>();
    }

    @Override
    public void registerRouter(String uri, Class<? extends Activity> activityClass) {
        activityRouterConfig.put(uri, activityClass);
    }

    @Override
    public Class<? extends Activity> getRouterActivity(String uri) {
        return activityRouterConfig.get(uri);
    }

    @Override
    public void registerRouter(String uri, IRouterProcess routerProcess) {
        routerProcessConfig.put(uri, routerProcess);
    }

    @Override
    public IRouterProcess getRouterProcess(String uri) {
        return routerProcessConfig.get(uri);
    }

    @Override
    public <T> void registerService(Class<T> serviceClass, Class<? extends T> implementClass) {
        serviceConfig.put(serviceClass, implementClass);
    }

    @Override
    public <T> Class<? extends T> getServiceImplementClass(Class<T> serviceClass) {
        return (Class<? extends T>) serviceConfig.get(serviceClass);
    }
}

File: app/src/main/java/cn/mycommons/androidmodular/app/ModuleConfig.kt
package cn.mycommons.androidmodular.app

import android.app.Activity
import cn.mycommons.modulebase.IModuleConfig
import cn.mycommons.modulebase.IRouterProcess
import java.util.*

/**
 * ModuleConfig <br></br>
 * Created by xiaqiulei on 2017-05-14.
 */
class ModuleConfig internal constructor() : IModuleConfig {

    private val activityRouterConfig: MutableMap<String, Class<out Activity>>
    private val routerProcessConfig: MutableMap<String, IRouterProcess>
    private val serviceConfig: MutableMap<Class<*>, Class<*>>

    init {
        activityRouterConfig = HashMap<String, Class<out Activity>>()
        routerProcessConfig = HashMap<String, IRouterProcess>()
        serviceConfig = HashMap<Class<*>, Class<*>>()
    }

    override fun registerRouter(uri: String, activityClass: Class<out Activity>) {
        activityRouterConfig[uri] = activityClass
    }

    override fun getRouterActivity(uri: String): Class<out Activity>? {
        return activityRouterConfig[uri]
    }

    override fun registerRouter(uri: String, routerProcess: IRouterProcess) {
        routerProcessConfig.put(uri, routerProcess)
    }

    override fun getRouterProcess(uri: String): IRouterProcess? {
        return routerProcessConfig[uri]
    }

    override fun <T> registerService(serviceClass: Class<T>, implementClass: Class<out T>) {
        serviceConfig.put(serviceClass, implementClass)
    }

    override fun <T> getServiceImplementClass(serviceClass: Class<T>): Class<out T> {
        return serviceConfig[serviceClass] as Class<out T>
    }
}